jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/diagram/ProcessInstanceDiagramPresenter.java

```diff
@@ -209,6 +210,20 @@ public void onDiagramNodeSelected(final String nodeId) {
         }
     }
 
+    public void onDiagramNodeSelectedByDblClick(final String nodeId) {
+        Optional<NodeInstanceSummary> nodeInstance = nodeInstances.stream().filter(node -> node.getNodeUniqueName().equals(nodeId)).findFirst();
```

Comment:
Not sure how to deal with multiple instances nodes .... we should let user to pick one they want.

It seems like we're getting only the first one which is IMO wrong.

Just an idea ... what if after double-click we present an element that would list all the instances that are applicable, so that user can pick one they want.

@cristianonicolai @lazarotti what do you think?

Reply:
I actually find double click not very user-friendly.  I would prefer to continue having a single click as a way to select the node. That triggers the side panel and could also show a sort of context menu to select from one to many instances to navigate to.